inspirehep/services/orcid_push/tasks.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Manage synchronization from INSPIRE to ORCID."""
+
+from __future__ import absolute_import, division, print_function
+
+from celery import shared_task
+
+from inspire_utils.logging import getStackTraceLogger
+
+from inspirehep.modules.orcid.api import push_record_with_orcid, get_author_putcodes
+from inspirehep.modules.orcid.utils import get_orcid_recid_key, redis_locking_context
+
+
+LOGGER = getStackTraceLogger(__name__)
+
+
+@shared_task(bind=True)
+def push_orcid(self, orcid, rec_id, oauth_token):
+    """Celery task to push a record to ORCID.
+
+    Args:
+        self(celery.Task): the task
+        orcid(string): an orcid identifier.
+        rec_id(int): inspire record's id to push to ORCID.
+        oauth_token(string): orcid token.
+    """
+    try:
+        attempt_push(orcid, rec_id, oauth_token)
+    except Exception:
+        raise self.retry(max_retries=3, countdown=300)
+
+
+def attempt_push(orcid, rec_id, oauth_token):
+    """Push a record to ORCID.
+
+    Args:
+        orcid(string): an orcid identifier.
+        rec_id(int): inspire record's id to push to ORCID.
+        oauth_token(string): orcid token.
+    """
+    put_code = get_putcode_from_redis(orcid, rec_id)
+
+    if not put_code:
+        record_put_codes, errors = get_author_putcodes(orcid, oauth_token)
```

Comment:
I don't see the point of returning `errors` only for the sake of logging. Isn't it more clear if `get_author_putcodes()` takes care of logging? Note that `push_record_with_orcid()` in the same module takes care of logging.